### DISCUSSION TITLE
Question about how Tunarr queries Plex

### DISCUSSION THREAD

**Discussion by FallenWyvern** [chrisbenincasa/tunarr#571] [Questions & Help] [answered]
I'm curious, does Tunarr query plex via the API found in the XML docs?
If so, why is there no option to filter shows by studio/platform?

Like if I use the url "http://PLEX_IP:32400/library/sections/1/all/?X-Plex-Token=MY_TOKEN" and look at the shows there, they have a studio field (and I know that the TV library can be filtered by Network)... can we not use those to help build channels?

1. Get all library items
2. Remove anything without the network we are filtering for...
3. Add everything that's left. 
4. Set to cyclical schedule.

Maybe store paths to each of these items (with their relevant metadata) in the database and you can even just have a "refresh" feature that checks each series for their total items, and in the case of a discrepency, updates the data.

Anyway this was supposed to be a question so I'll leave it there... does Tunarr use this field?

**Comment by chrisbenincasa** [comment]
Hey there! I am able to see and filter on studio for both movies and TV shows:

![Screenshot from 2024-06-25 16-27-31](https://github.com/chrisbenincasa/tunarr/assets/1640671/f3272c64-54a8-40ac-8be2-1d7c7e6fabba)


![Screenshot from 2024-06-25 16-28-43](https://github.com/chrisbenincasa/tunarr/assets/1640671/087356e2-4067-400e-bc56-01405fd3cf87)

Are you not able to see that filter option in the dropdown?

![image](https://github.com/chrisbenincasa/tunarr/assets/1640671/6e846a25-45d7-4bea-ba27-b1a79300cb07)

Tunarr uses the same endpoints that Plex does to collect available filter fields and values from your server.

```
/library/sections/{ID}/all?includeMeta=1&includeAdvanced=1&X-Plex-Container-Start=0&X-Plex-Container-Size=0

**Comment by FallenWyvern** [comment]
Nope but I have a semi botched install. I'll try to clean this up and see
if that's the same after

**Reply by chrisbenincasa** [reply, marked as answer]
Sounds good. You can check against your Plex server using that endpoint too

**Reply by FallenWyvern** [reply]
A reinstall of the Tunarr docker, and deleting of the AppData, restored the filter functionality.
For the record the filter was there, but it wasn't filtering anything. I'm not sure what kind of bug that would need to be reported as but if it happens again, I'll grab the logs and report it.

**Reply by chrisbenincasa** [reply]
Glad it's sorted now. Definitely strange that you had to do a fresh install, since none of the data related to filtering Plex or pulling data in that view OSS persisted anywhere. Definitely let me know if it happens again and we can dig in further.
